src/Utils.cc

```diff
@@ -563,4 +574,271 @@ thread_type_t fbgemmGetThreadPartition(
   return th_info;
 }
 
+namespace {
+
+// implementation taken from pytorch/c10/util/llvmMathExtras.h
+template <typename T>
+size_t count_leading_zeros(T val) {
+  if (!val)
+    return std::numeric_limits<T>::digits;
+
+  size_t zero_bits = 0;
+  for (T shift = std::numeric_limits<T>::digits >> 1; shift; shift >>= 1) {
+    T tmp = val >> shift;
+    if (tmp)
+      val = tmp;
+    else
+      zero_bits |= shift;
+  }
+  return zero_bits;
+}
+
+// histogram size per thread
+constexpr int RDX_HIST_SIZE = 256;
+
+#define COMBINE_PREFIX_SUM_IN_RANGE(                            \
+    sum, prev_sum, bins_beg, bins_end, hist, hist_ps, nthreads) \
+  for (int bins = bins_beg; bins < bins_end; ++bins) {          \
+    for (int t = 0; t < nthreads; ++t) {                        \
+      sum += histogram[t * RDX_HIST_SIZE + bins];               \
+      histogram_ps[t * RDX_HIST_SIZE + bins] = prev_sum;        \
+      prev_sum = sum;                                           \
+    }                                                           \
+  }
+
+void combine_prefix_sum(
+    int nthreads,
+    int elements_count,
+    int* histogram,
+    int* histogram_ps) {
+  int sum = 0, prev_sum = 0;
+  COMBINE_PREFIX_SUM_IN_RANGE(
+      sum, prev_sum, 0, RDX_HIST_SIZE, histogram, histogram_ps, nthreads);
+
+  histogram_ps[RDX_HIST_SIZE * nthreads] = prev_sum;
+  // TODO(dszwicht): Is assert sufficient? In most cases, it will work only in
+  // debug build.
+  assert(prev_sum == elements_count);
+  // Suppress unused variable warning
+  (void)elements_count;
```

Comment:
Commenting out `elements_count` in the arguments means these lines become unnecessary.

Reply:
Please see [above](https://github.com/pytorch/FBGEMM/pull/1672#discussion_r1160502545) comment.